src/code.jl

```diff
@@ -737,7 +737,16 @@ end
 
 ### Common subexprssion evaluation
 
-@inline newsym(::Type{T}) where T = Sym{T}(gensym("cse"))
+"""
+    newsym!(state::CSEState, ::Type{T})
+
+Generates new symbol of type `T` with unique name in `state`.
+"""
+@inline function newsym!(state, ::Type{T}) where T 
+    name = "##cse#$(state.varid[])"
+    state.varid[] += 1
+    Sym{T}(Symbol(name))
+end
```

Comment:
What if a user maliciously creates a `Sym` with name `##cse#1`?

Reply:
We're not really building a secure software. There's already a lot of ways users can make our stuff break if they look into internals, so it doesn't feel worth to try and guard against this.